src/pymor/discretizers/dunegdt/cg.py

```diff
@@ -0,0 +1,565 @@
+from pymor.core.config import config, is_jupyter
+
+
+if config.HAVE_DUNEGDT:
+    import numpy as np
+    from functools import partial
+    from numbers import Number
+
+    from dune.xt.grid import (
+            ApplyOnBoundaryIntersections,
+            ApplyOnCustomBoundaryIntersections,
+            Dim,
+            DirichletBoundary,
+            NeumannBoundary,
+            RobinBoundary,
+            Walker,
+            )
+    from dune.xt.functions import divergence, GridFunction as GF
+    from dune.xt.la import Istl
+    from dune.gdt import (
+            ContinuousLagrangeSpace,
+            DirichletConstraints,
+            DiscontinuousLagrangeSpace,
+            DiscreteFunction,
+            LocalElementIntegralBilinearForm,
+            LocalElementIntegralFunctional,
+            LocalElementProductIntegrand,
+            LocalIntersectionIntegralBilinearForm,
+            LocalIntersectionIntegralFunctional,
+            LocalIntersectionNormalComponentProductIntegrand,
+            LocalIntersectionProductIntegrand,
+            LocalLaplaceIntegrand,
+            LocalLinearAdvectionIntegrand,
+            MatrixOperator,
+            VectorFunctional,
+            boundary_interpolation,
+            default_interpolation,
+            make_element_sparsity_pattern,
+            )
+
+    from pymor.algorithms.timestepping import ExplicitEulerTimeStepper, ImplicitEulerTimeStepper
+    from pymor.analyticalproblems.elliptic import StationaryProblem
+    from pymor.analyticalproblems.instationary import InstationaryProblem
+    from pymor.analyticalproblems.functions import Function, ConstantFunction, LincombFunction
+    from pymor.bindings.dunegdt import (
+            DuneGDT1dasNumpyVisualizer,
+            DuneGDT1dMatplotlibVisualizer,
+            DuneGDTK3dVisualizer,
+            DuneGDTParaviewVisualizer,
+            DuneXTMatrixOperator,
+            DuneXTVector,
+            DuneXTVectorSpace,
+            )
+    from pymor.core.base import ImmutableObject
+    from pymor.discretizers.dunegdt.domaindiscretizers.default import discretize_domain_default
+    from pymor.models.basic import InstationaryModel, StationaryModel
+    from pymor.operators.constructions import ConstantOperator, LincombOperator, VectorArrayOperator
+    from pymor.tools.floatcmp import float_cmp
+
+
+    def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretizer=None,
+                                 grid_type=None, grid=None, boundary_info=None,
+                                 order=1, data_approximation_order=2, la_backend=Istl(),
```

Comment:
Is that a mutable default for `la_backend`?

Reply:
Its a default for `la_backend`, what do you mean by mutable here?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Thx, did not know that. Should not be a problem here.